File: clients/javascript/src/resources/network.ts
import type { HieroProxyClient } from "../client";
import type {
  ExchangeRatesResponse,
  NetworkFeeResponse,
  NetworkStakeResponse,
  NetworkSuppliesResponse,
} from "./types";

/**
 * Network resource — read-only queries for exchange rates, fees, staking, and supply.
 */
export class NetworkResource {
  constructor(private readonly client: HieroProxyClient) {}

  /**
   * Get current and next HBAR/USD exchange rates.
   */
  async getExchangeRates(): Promise<ExchangeRatesResponse> {
    return this.client.get("/api/v1/network/exchange-rates");
  }

  /**
   * Get EVM gas fees for smart contract operations.
   */
  async getFees(): Promise<NetworkFeeResponse[]> {
    return this.client.get("/api/v1/network/fees");
  }

  /**
   * Get network staking configuration and statistics.
   */
  async getStake(): Promise<NetworkStakeResponse> {
    return this.client.get("/api/v1/network/stake");
  }

  /**
   * Get HBAR supply figures (released and total).
   */
  async getSupplies(): Promise<NetworkSuppliesResponse> {
    return this.client.get("/api/v1/network/supplies");
  }
}

File: clients/python/src/hiero_proxy/resources/network.py
"""Network resource."""

from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from hiero_proxy.client import AsyncHieroProxyClient, HieroProxyClient


class NetworkResource:
    """Network queries — exchange rates, fees, staking, supply."""

    def __init__(self, client: HieroProxyClient | AsyncHieroProxyClient) -> None:
        self._client = client

    def get_exchange_rates(self) -> Any:
        return self._client.get("/api/v1/network/exchangerates")

    def get_fees(self) -> Any:
        return self._client.get("/api/v1/network/fees")

    def get_staking_info(self) -> Any:
        return self._client.get("/api/v1/network/staking")

    def get_supply(self) -> Any:
        return self._client.get("/api/v1/network/supply")
